File: mybatis/src/main/java/data/spring/mybatis/application/required/ProductUseCase.java
package data.spring.mybatis.application.required;

import data.spring.mybatis.application.service.command.ProductSearchCommand;
import data.spring.mybatis.application.service.command.ProductUpdateCommand;
import data.spring.mybatis.domain.Product;

import java.util.List;
import java.util.Optional;

public interface ProductUseCase {
    List<Product> findAll(ProductSearchCommand searchCommand);
    void save(Product product);
    int saveAll(List<Product> products);
    Optional<Product> findById(Long productId);
    int updateAll(List<ProductUpdateCommand> updateBatchRequest);
}

File: mybatis/src/main/java/data/spring/mybatis/application/provided/product/ProductUseCase.kt
package data.spring.mybatis.application.provided.product

import data.spring.mybatis.application.service.product.command.ProductSearchCommand
import data.spring.mybatis.application.service.product.command.ProductUpdateCommand
import data.spring.mybatis.domain.product.Product
import java.util.*

interface ProductUseCase : ProductUseCaseEx {
    fun findById(productId: Long): Product?

    fun findWithCond(searchCommand: ProductSearchCommand): List<Product>

    fun save(product: Product)

    fun saveAll(products: List<Product>): Int
}